src/Strategy/NuGet/ProjectAssetsJson.hs

```diff
@@ -68,52 +78,165 @@ analyze' file = do
       , dependencyManifestFiles = [file]
       }
 
-newtype ProjectAssetsJson = ProjectAssetsJson
-  { targets :: Map.Map Text (Map.Map Text DependencyInfo)
+data NuGetDep = NuGetDep
+  { depName :: Text
+  , depVersion :: Text
+  , completeDepType :: Text
+  , completeDeepDeps :: Map.Map Text Text
   }
-  deriving (Show)
+  deriving (Show, Eq, Ord)
 
-instance FromJSON ProjectAssetsJson where
-  parseJSON = withObject "ProjectAssetsJson" $ \obj ->
-    ProjectAssetsJson <$> obj .: "targets"
+data ProjectAssetsJson = ProjectAssetsJson
+  { targets :: Map.Map Text (Map.Map Text DependencyInfo)
+  , projectFramework :: Map.Map Text (Set Text)
+  }
+  deriving (Show, Eq, Ord)
 
 data DependencyInfo = DependencyInfo
   { depType :: Text
   , deepDeps :: Map.Map Text Text
   }
-  deriving (Show)
+  deriving (Show, Eq, Ord)
 
 instance FromJSON DependencyInfo where
   parseJSON = withObject "Dependency" $ \obj ->
     DependencyInfo <$> obj .: "type"
       <*> obj .:? "dependencies" .!= Map.empty
 
-data NuGetDep = NuGetDep
-  { depName :: Text
-  , depVersion :: Text
-  , completeDepType :: Text
-  , completeDeepDeps :: Map.Map Text Text
-  }
-  deriving (Show)
+instance FromJSON ProjectAssetsJson where
+  parseJSON = withObject "ProjectAssetsJson" $ \obj -> do
+    targets <- obj .: "targets"
+    projectFrameworks <- obj .: "project" |> "frameworks"
+    deps <- parseFramework projectFrameworks
+    pure $ ProjectAssetsJson targets deps
+    where
+      (|>) :: FromJSON a => Parser Object -> Text -> Parser a
+      (|>) parser key = do
+        obj <- parser
+        obj .: key
+
+      parseFrameworkDeps :: Value -> Parser (Set Text)
+      parseFrameworkDeps = withObject "parseFrameworkDeps" $ \o -> do
+        depsObj :: Maybe Object <- o .:? "dependencies"
+        deps <- case depsObj of
+          Nothing -> pure []
+          Just hm -> pure (fst <$> HM.toList hm)
+        pure $ Set.fromList deps
+
+      parseFramework :: Value -> Parser (Map.Map Text (Set Text))
+      parseFramework = withObject "parseFramework" $ \o -> do
+        projectFrameworks <- for (HM.toList o) $ \(framework, kv) -> do
+          frameworkDeps <- parseFrameworkDeps kv
+          pure (framework, frameworkDeps)
+        pure $ Map.fromList projectFrameworks
 
 buildGraph :: ProjectAssetsJson -> Graphing Dependency
-buildGraph project = unfold direct deepList toDependency
+buildGraph project = foldr (<>) Graphing.empty graphsOfTargetFrameworks
+  where
+    graphsOfTargetFrameworks =
+      map
+        (graphOfFramework $ projectFramework project)
+        (Map.toList $ targets project)
+
+graphOfFramework :: Map.Map Text (Set Text) -> (Text, Map.Map Text DependencyInfo) -> Graphing Dependency
+graphOfFramework projectFrameworkDeps (targetFramework, targetFrameworkDeps) =
+  withoutTags
+    . withoutProjects
+    . promoteToDirect isDirectDep
+    $ unfoldDeep allResolvedDeps getTransitiveDeps toDependency
```

Comment:
The way I view the `Dependency` type is that once we've built something of type `Dependency`, we generally shouldn't touch it ever again

The fields on `Dependency` have very few guarantees, and depending on the tags field here to communicate type information indirectly feels brittle

When we need to modify `Dependency` values, it's usually a sign that we can model the domain more closely by using an intermediate type

We almost do that here with the `NuGetDep` type, but we eagerly convert it to `Dependency` with `toDependency`. May I recommend instead:

```hs
unfoldDeep allResolvedDeps getTransitiveDeps id :: Graphing NuGetDep
```

do the `Graphing.shrink withoutProjects` on that, slightly changing `withoutProjects`

```hs
withoutProjects :: Graphing NuGetDep -> Graphing NuGetDep
```

_then_ turn it into `Graphing Dependency` with `Graphing.gmap toDependency`